http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -856,15 +865,22 @@ class DataprocInterpreter[F[_]: Parallel](
     }
   }
 
-  private def getSoftwareConfig(googleProject: GoogleProject,
-                                runtimeName: RuntimeName,
-                                machineConfig: RuntimeConfig.DataprocConfig
+  // This file is worth keeping track of. It's the source of truth for hail machine provisioning
+  // https://github.com/hail-is/hail/blob/75f351d43a6c6b2e87e7d37531be63aaf11af9df/hail/python/hailtop/hailctl/dataproc/start.py#L20
+  def getSoftwareConfig(googleProject: GoogleProject,
+                        runtimeName: RuntimeName,
+                        machineConfig: RuntimeConfig.DataprocConfig,
+                        jupyterResourceConstraints: RuntimeResourceConstraints
   ): SoftwareConfig = {
     val dataprocProps = if (machineConfig.numberOfWorkers == 0) {
       // Set a SoftwareConfig property that makes the cluster have only one node
       Map("dataproc:dataproc.allow.zero.workers" -> "true")
     } else Map.empty[String, String]
 
+    val memoryLimitInMb =
+      (jupyterResourceConstraints.totalMachineMemory.bytes - jupyterResourceConstraints.memoryLimit.bytes) / 1048576
```

Comment:
what is 1048576? Maybe a comment?

Reply:
Added into MemorySize. LMK if you have a better suggestion for the name of the var.